client/src/components/Leaderboard.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import Paper from '@material-ui/core/Paper';
+import TableCell from '@material-ui/core/TableCell';
+import TableRow from '@material-ui/core/TableRow';
+import MaterialTable from './MaterialTable';
+
+const styles = theme => ({
+  root: {
+    width: '50%',
+    minWidth: 200,
+    margin: `${theme.spacing.unit * 2}px auto`,
+    padding: theme.spacing.unit * 2,
+  },
+  title: {
+    fontSize: theme.typography.h5.fontSize,
+  },
+});
+
+class Leaderboard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      leaderboards: [],
+    };
+  }
+
+  componentDidMount() {
+    const { getTopFive } = this.props;
+    getTopFive()
+      .then(({ data: leaderboards }) => {
+        this.setState({
+          leaderboards,
+        });
+      });
+  }
+
+  render() {
+    const { classes } = this.props;
+    const { leaderboards } = this.state;
+    if (leaderboards.length) {
+      const leaderboardsRows = leaderboards.map(leader => (
+        <TableRow key={leader.username}>
```

Comment:
`key` should identify the content of the row. I think key should include both username and score. so, if one of them changes, the row is re-rendered.

I personally just do JSON.stringify of the content to be rendered, but there might be better approaches.

Reply:
I think I will keep it like this for now.